tests/set_namespace_go.sh

```diff
@@ -41,17 +41,18 @@ metadata:
 data:
   "namespace": "example-ns"
 EOF
-kpt fn run .
-assert_contains_string example-configs/gatekeeper.yaml "namespace: example-ns"
+kpt fn source example-configs |
+  kpt fn run --fn-path fc.yaml >out.yaml
+assert_contains_string out.yaml "namespace.*:.*example-ns"
 
 # Use the linux executable to test using kpt exec runtime on linux
 testcase "kpt_set_namespace_go_exec_imperative_linux"
 kpt pkg get "$SDK_REPO"/example-configs example-configs
 kpt pkg get "$CATALOG_REPO"/functions/go ./
 kpt fn run example-configs --enable-exec --exec-path "$(pwd)"/go/set-namespace/set-namespace-linux -- namespace=example-ns
```

Comment:
- We should get rid of any non-containerized tests, also generally standardize the tests as part of support for curated functions. They're pretty adhoc and hack right now.
- Should test for idempotency and other properties as discusse don the doc.

@mengqiy

Reply:
The test files used in e2e test, (`"$SDK_REPO"/example-configs`) cannot keep idempotency. Because after applying the namespace function, there will be multiple resources with duplicate ID (group, kind, version, namespace and name tuple). I will use a subdirectory (`"$SDK_REPO"/example-configs/shipping-dev`) in the test.